src/solutions/ode_solutions.jl

```diff
@@ -15,9 +15,68 @@ struct ODESolution{T,N,uType,uType2,DType,tType,rateType,P,A,IType,DE} <: Abstra
   destats::DE
   retcode::Symbol
 end
-(sol::ODESolution)(t,deriv::Type=Val{0};idxs=nothing,continuity=:left) = sol.interp(t,idxs,deriv,sol.prob.p,continuity)
+(sol::ODESolution)(t,deriv::Type=Val{0};idxs=nothing,continuity=:left) = sol(t,deriv,idxs,continuity)
 (sol::ODESolution)(v,t,deriv::Type=Val{0};idxs=nothing,continuity=:left) = sol.interp(v,t,idxs,deriv,sol.prob.p,continuity)
 
+function (sol::ODESolution)(t::Real,::Type{Val{0}},idxs::Nothing,continuity)
+  sol.interp(t,idxs,Val{0},sol.prob.p,continuity)
+end
+
+function (sol::ODESolution)(t::AbstractVector{<:Real},::Type{Val{0}},idxs::Nothing,continuity)
+  augment(sol.interp(t,idxs,Val{0},sol.prob.p,continuity), sol)
+end
+
+function (sol::ODESolution)(t::Real,::Type{Val{0}},idxs::Integer,continuity)
+  sol.interp(t,idxs,Val{0},sol.prob.p,continuity)
+end
+function (sol::ODESolution)(t::Real,::Type{Val{0}},idxs::AbstractVector{<:Integer},continuity)
+  sol.interp(t,idxs,Val{0},sol.prob.p,continuity)
+end
+function (sol::ODESolution)(t::AbstractVector{<:Real},::Type{Val{0}},idxs::Integer,continuity)
+  sol.interp(t,idxs,Val{0},sol.prob.p,continuity)
+end
+function (sol::ODESolution)(t::AbstractVector{<:Real},::Type{Val{0}},idxs::AbstractVector{<:Integer},continuity)
+  sol.interp(t,idxs,Val{0},sol.prob.p,continuity)
+end
+
+function (sol::ODESolution)(t::Real,::Type{Val{0}},idxs,continuity)
+  if !issymbollike(idxs)
+    error("Incorrect specification of `idxs`")
+  end
+  augment(sol.interp([t],nothing,Val{0},sol.prob.p,continuity), sol)[idxs]
+end
+
+function (sol::ODESolution)(t::Real,::Type{Val{0}},idxs::AbstractVector,continuity)
+  if any(.!issymbollike.(idxs))
+    error("Incorrect specification of `idxs`")
+  end
+  interp_sol = augment(sol.interp([t],nothing,Val{0},sol.prob.p,continuity), sol)
+  [first(interp_sol[idx]) for idx in idxs]
+end
+
+function (sol::ODESolution)(t::AbstractVector{<:Real},::Type{Val{0}},idxs,continuity)
+  if !issymbollike(idxs)
+    error("Incorrect specification of `idxs`")
+  end
+  interp_sol = augment(sol.interp(t,nothing,Val{0},sol.prob.p,continuity), sol)
+  DiffEqArray(interp_sol[idxs], t)
+end
+
+function (sol::ODESolution)(t::AbstractVector{<:Real},::Type{Val{0}},idxs::AbstractVector,continuity)
+  if any(.!issymbollike.(idxs))
+    error("Incorrect specification of `idxs`")
+  end
+  interp_sol = augment(sol.interp(t,nothing,Val{0},sol.prob.p,continuity), sol)
+  u = [[interp_sol[idx][i] for idx in idxs] for i in 1:length(t)]
+  DiffEqArray(u, t)
+end
+
+for T in 1:3
+  function (sol::ODESolution)(t,::Type{Val{T}},idxs,continuity)
+    error("Higher-order interpolation is not implemented.")
+  end
+end
```

Comment:
Is this approach okay to handle higher-order interpolation? @devmotion

Reply:
Just don't define these functions?

Reply:
I guess the idea was to have a user-readable error. @YingboMa Any thoughts?

Reply:
How about
```julia
function (sol::ODESolution)(t::AbstractVector{<:Real},::Type{Val{N}},idxs::AbstractVector,continuity) where N
    N == 0 || error("Higher-order interpolation is not implemented.")
```

Reply:
@YingboMa I think erroring out at `sol(t, idxs=[...])` stage for higher-order interpolation is wrong (as @devmotion already mentioned). I just realized that it has been already implemented in many scenarios.

https://github.com/SciML/SciMLBase.jl/blob/master/src/interpolation.jl#L274-L314
https://github.com/SciML/SciMLBase.jl/blob/master/src/interpolation.jl#L335-L369
https://github.com/SciML/SciMLBase.jl/blob/master/src/interpolation.jl#L387
https://github.com/SciML/SciMLBase.jl/blob/master/src/interpolation.jl#L414
https://github.com/SciML/SciMLBase.jl/blob/master/src/interpolation.jl#L439
https://github.com/SciML/SciMLBase.jl/blob/master/src/interpolation.jl#L465

Reply:
I have set a default at `interpolant` stage https://github.com/SciML/SciMLBase.jl/pull/66/commits/b97346ced028c6113082c0cb83ae493047c43e6a .